 dropzonejs.install |  3 ---
 dropzonejs.module  | 12 ++++--------
 2 files changed, 4 insertions(+), 11 deletions(-)

diff --git a/dropzonejs.install b/dropzonejs.install
index 4046902..6d8d625 100644
--- a/dropzonejs.install
+++ b/dropzonejs.install
@@ -22,9 +22,6 @@ function dropzonejs_requirements($phase) {
   }
   else {
     $path = DRUPAL_ROOT . '/libraries/dropzone/dist/min/dropzone.min.js';
-    if (\Drupal::moduleHandler()->moduleExists('libraries')) {
-      $path = libraries_get_path('dropzone') . '/dist/min/dropzone.min.js';
-    }
 
     // Is the library found in the root libraries path.
     $library_found = file_exists($path);
diff --git a/dropzonejs.module b/dropzonejs.module
index b6df253..09e55ef 100644
--- a/dropzonejs.module
+++ b/dropzonejs.module
@@ -78,11 +78,7 @@ function dropzonejs_library_info_build() {
     /** @var \Drupal\Core\Asset\LibrariesDirectoryFileFinder $library_file_finder */
     $library_file_finder = \Drupal::service('library.libraries_directory_file_finder');
     $exif_path = $library_file_finder->find('exif-js/exif.js');
-  }
-  elseif (\Drupal::moduleHandler()->moduleExists('libraries')) {
-    $exif_path = libraries_get_path('exif-js') . '/exif.js';
-  }
-  else {
+  } else {
     $exif_path = DRUPAL_ROOT . '/libraries/exif-js/exif.js';
   }
 
@@ -111,10 +107,10 @@ function dropzonejs_library_info_build() {
  */
 function dropzonejs_library_info_alter(&$libraries, $extension) {
   if ($extension == 'dropzonejs' && \Drupal::moduleHandler()->moduleExists('libraries')) {
-    $libraries['dropzonejs']['js'] = ['/' . libraries_get_path('dropzone') . '/dist/min/dropzone.min.js' => []];
-    $libraries['dropzonejs']['css']['component'] = ['/' . libraries_get_path('dropzone') . '/dist/min/dropzone.min.css' => []];
+    $libraries['dropzonejs']['js'] = ['/libraries/dropzone/dist/min/dropzone.min.js' => []];
+    $libraries['dropzonejs']['css']['component'] = ['/libraries/dropzone/dist/min/dropzone.min.css' => []];
 
-    if ($exif_path = libraries_get_path('exif-js')) {
+    if ($exif_path = DRUPAL_ROOT . '/libraries/exif-js/exif.js') {
       $libraries['exif-js']['js'] = ['/' . $exif_path . '/exif.js' => []];
     }
   }


