diff --git a/vendor/h5p/h5p-editor/h5peditor.class.php b/vendor/h5p/h5p-editor/h5peditor.class.php
index c1a3fb5a..37a60a58 100644
--- a/vendor/h5p/h5p-editor/h5peditor.class.php
+++ b/vendor/h5p/h5p-editor/h5peditor.class.php
@@ -373,7 +373,7 @@ public function findEditorLibraries($machineName, $majorVersion, $minorVersion)
    *
    * @return array Libraries that was requested
    */
-  public function getLibraryData($machineName, $majorVersion, $minorVersion, $languageCode, $prefix = '', $fileDir = '', $defaultLanguage) {
+  public function getLibraryData($machineName, $majorVersion, $minorVersion, $languageCode, $prefix = '', $fileDir = '', $defaultLanguage = '') {
     $libraryData = new stdClass();

     $library = $this->h5p->loadLibrary($machineName, $majorVersion, $minorVersion);
