diff --git a/src/Controller/SamlController.php b/src/Controller/SamlController.php
index 4df45ac..20320c2 100644
--- a/src/Controller/SamlController.php
+++ b/src/Controller/SamlController.php
@@ -148,7 +148,14 @@ class SamlController extends ControllerBase {
     // code... so it wouldn't need to be executed inside a render context. The
     // standard exception handling does, though.
     $function = function () {
-      return $this->saml->login($this->getUrlFromDestination());
+      $destination = \Drupal::request()->query->get('destination');
+      if (isset($destination)) {
+        return $this->saml->login($this->getUrlFromDestination());
+      } else {
+        $request = $this->requestStack->getCurrentRequest();
+        $destination_url = $request->headers->get('referer');
+        return $this->saml->login($destination_url);
+      }
     };
     // This response redirects to an external URL in all/common cases. We count
     // on the routing.yml to specify that it's not cacheable.
