diff --git src/TwigTweakExtension.php src/TwigTweakExtension.php
index d007798..072c953 100644
--- src/TwigTweakExtension.php
+++ src/TwigTweakExtension.php
@@ -703,9 +703,7 @@ class TwigTweakExtension extends AbstractExtension {
     ob_start();
     // phpcs:ignore Drupal.Functions.DiscouragedFunctions.Discouraged
     print eval($code);
-    $output = ob_get_contents();
-    ob_end_clean();
-    return $output;
+    return ob_get_clean();
   }

}
